Composer/packages/client/src/components/CreationFlow/v2/DefineConversation.tsx

```diff
@@ -0,0 +1,369 @@
+        }
+      },
+    },
+    description: {
+      required: false,
+    },
+    primaryLanguage: {
+      required: false,
+    },
+    runtimeChoice: {
+      required: false,
+    },
+    schemaUrl: {
+      required: false,
+    },
+    location: {
+      required: true,
+      defaultValue:
+        focusedStorageFolder !== null && Object.keys(focusedStorageFolder as Record<string, any>).length
+          ? Path.join(focusedStorageFolder.parent, focusedStorageFolder.name)
+          : '',
+    },
+  };
+  const { formData, formErrors, hasErrors, updateField, updateForm, validateForm } = useForm(formConfig);
+  const [isImported, setIsImported] = useState<boolean>(false);
+
+  useEffect(() => {
+    if (props.location?.state) {
+      const { imported } = props.location.state;
+      setIsImported(imported);
+    }
+  }, [props.location?.state]);
+
+  useEffect(() => {
+    const formData: DefineConversationFormData = {
+      name: getDefaultName(),
+      primaryLanguage: defaultPrimaryLanguage,
+      runtimeChoice: defaultRuntime,
+      description: '',
+      schemaUrl: '',
+      location:
+        focusedStorageFolder !== null && Object.keys(focusedStorageFolder as Record<string, any>).length
+          ? Path.join(focusedStorageFolder.parent, focusedStorageFolder.name)
+          : '',
+    };
+    if (props.location?.search) {
+      const decoded = decodeURIComponent(props.location.search);
+      const { name, description, schemaUrl } = querystring.parse(decoded);
+      if (description) {
+        formData.description = description as string;
+      }
+
+      if (schemaUrl) {
+        formData.schemaUrl = schemaUrl as string;
+      }
+
+      if (name) {
+        formData.name = name as string;
+      } else {
+        formData.name = getDefaultName();
+      }
+    }
+    updateForm(formData);
+  }, [templateId]);
+
+  useEffect(() => {
+    validateForm();
+  }, [focusedStorageFolder]);
+
+  const handleSubmit = useCallback(
+    (e) => {
+      e.preventDefault();
+      if (hasErrors) {
+        return;
+      }
+
+      // handle extra properties in the case of an imported bot project
+      const dataToSubmit = {
+        ...formData,
+      };
+      if (props.location?.state) {
+        const { alias, eTag, imported, templateDir, urlSuffix } = props.location.state;
+
+        if (imported) {
+          dataToSubmit.templateDir = templateDir;
+          dataToSubmit.eTag = eTag;
+          dataToSubmit.urlSuffix = urlSuffix;
+          dataToSubmit.alias = alias;
+          dataToSubmit.preserveRoot = true;
+
+          // create a notification to indicate import success
+          const notification = createNotification({
+            type: 'success',
+            title: '',
+            onRenderCardContent: ImportSuccessNotificationWrapper({
+              importedToExisting: false,
+            }),
+          });
+          addNotification(notification);
+        }
+      }
+
+      onSubmit(
+        {
```

Comment:
does prettier format this object parameter like it's checked-in? why not `{ ...dataToSubmit },`

Reply:
Handled